Core/NoOpRunner.Core/Shapes/BaseShape.cs

```diff
@@ -98,5 +97,14 @@ public virtual void OnClick()
                 x.Color = Color.Green;
             });
         }
+
+        public abstract bool CanOverlap(BaseShape other);
+        public abstract void OnCollision(BaseShape other);
+
+        public virtual void DoDamage(int damage) { }
+
+        public virtual void DoHeal(int heal) { }
```

Comment:
These don't belong here, staticShapes neither damage nor heal. Move to EntityShape, and maybe merge into one (maybe `modifyHealth`? Will likely have to refactor anyways when we start implementing the second player)

Reply:
Removed this code from BaseShape.cs. Did not move that code to EntityShape for now.